python/az/aro/azext_aro/custom.py

```diff
@@ -195,18 +197,43 @@ def aro_list_credentials(client, resource_group_name, resource_name):
     return client.list_credentials(resource_group_name, resource_name)
 
 
-def aro_list_admin_credentials(cmd, client, resource_group_name, resource_name):
+def aro_list_admin_credentials(cmd, client, resource_group_name, resource_name,
+                               overwrite=False, file="kubeconfig", internal=False):
     # check for the presence of the feature flag and warn
     # the check shouldn't block the API call - ARM can cache a feature state for several minutes
     feature_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_RESOURCE_FEATURES)
     feature = feature_client.features.get(resource_provider_namespace="Microsoft.RedHatOpenShift",
                                           feature_name="AdminKubeconfig")
-    accepted_states = [feature_client.features.models.SubscriptionFeatureRegistrationState.REGISTERED,
-                       feature_client.features.models.SubscriptionFeatureRegistrationState.REGISTERING]
+    accepted_states = ["Registered",
+                       "Registering"]
     if feature.properties.state not in accepted_states:
         logger.warning("This operation requires the Microsoft.RedHatOpenShift/AdminKubeconfig feature to be registered")
         logger.warning("To register run: az feature register --namespace Microsoft.RedHatOpenShift -n AdminKubeconfig")
-    return client.list_admin_credentials(resource_group_name, resource_name)
+    query_result = client.list_admin_credentials(resource_group_name, resource_name)
+    file_mode = "x"
+    if overwrite:
+        file_mode = "w"
+    yaml_data = b64decode(query_result.kubeconfig).decode('UTF-8')
+    if not internal:
+        yaml_data = _external_kubeconfig(yaml_data)
+    try:
+        with open(file, file_mode) as f:
+            f.write("apiVersion: v1\n")
+            f.write(yaml_data)
+    except FileExistsError:
+        logger.error("File %s already exists, use the --overwrite parameter to overwrite file.", file)
+    logger.info("Kubeconfig written to file: %s", file)
+
+
+def _external_kubeconfig(yaml_data):
+    kubeconfig = yaml.safe_load(yaml_data)
+    try:
+        for cluster in kubeconfig["clusters"]:
+            cluster["cluster"].pop("certificate-authority-data")  # no custom CA required for external API
+            cluster["cluster"]["server"] = cluster["cluster"]["server"].replace("https://api-int.", "https://api.")
+    except KeyError:
+        logger.error("Unexpected Kubeconfig format. Please contact Azure Red Hat OpenShift support.")
+    return yaml.dump(kubeconfig)
```

Comment:
Discussed on a call. Posting for visibility:

* Check if we can use `api.` instead of `api-int.` for all cluster (public, private)
* If we can - consider modifying kubeconfig on the RP side instead of the cli. Let’s keep CLI simple (get a thing & dump it somewhere)